src/eth/provider.ts

```diff
@@ -33,6 +32,10 @@ export async function getEthereumProvider(
     }
   }
 
+  if (connection.isConnected()) {
```

Comment:
This validation prevents the provider `magic` from losing the connection and enters a login/logout loop